### ISSUE TITLE
Need param to stop run if condition holds True?

### ISSUE THREAD

**Issue by mikefenton** [jmmcd/PonyGE2#37]
Might add in a param to stop a run if a condition is true, e.g. if a string-match fitness gets to 0 (i.e. a perfect match), rather than just continuing on pointlessly for the required total number of generations.

**Comment by mikefenton** [comment]
Actually this is just a different search loop.

**Comment by dvpfagan** [comment]
You actually hit on an interesting topic.

Traditionally GE/DP would run until the solution was found or your max
generations.

In problems like regression where a 0 value solution is tough to find the
idea of a hit value was used so if a fitness of say 0.05 was achieved this
would be counted as zero.

So the debate that comes about from this is how do we deal with runs that
finish after say 20 gens and others that finish after 50 gens. To compare
the two we need equal amounts of data so do we just keep running till 50
gens or after we reach a solution do we just duplicate the last row of data
to fill up the missing generations.

Answers on a postcard :D

**Comment by mikefenton** [comment]
For applications where we don't want to compare runs but just want to find a perfect solution, it's pointless to continue past the point where a solution has been found. E.g. string match.

**Comment by dvpfagan** [comment]
agreed but if you want to compare 30 runs and get averages you either need
to run for the same number of generations of else duplicate the last line
of the stats file.

**Comment by jmmcd** [comment]
Decision: worth doing longer-term, not that bothered right now.

Implement with `--termination-threshold` and check for existence in the main search loop, and check for best fitness `<=` or `>=` depending on maximise or minimise.
